pkgs/games/etlegacy/default.nix

```diff
@@ -0,0 +1,124 @@
+{ stdenv
+, lib
+, makeWrapper
+, writeScriptBin
+, fetchgit
+, fetchFromGitHub
+, fetchzip
+, fetchurl
+, runCommand
+, cmake
+, git
+, glew
+, SDL2
+, zlib
+, minizip
+, libjpeg
+, curl
+, lua5_4
+, libogg
+, libtheora
+, freetype
+, libpng
+, sqlite
+, openal
+, unzip
+, cjson
+}:
+
+let
+  version = "2.81.1";
+  pkgname = "etlegacy";
+  lua = lua5_4;
+  mirror = "https://mirror.etlegacy.com";
+  fetchAsset = {asset, sha256}: fetchurl
+    { url = mirror + "/etmain/" + asset;
+    inherit sha256;
+    };
+  pak0 = fetchAsset
+    { asset = "pak0.pk3";
+    sha256 = "712966b20e06523fe81419516500e499c86b2b4fec823856ddbd333fcb3d26e5";
+    };
+  pak1 = fetchAsset
+    { asset = "pak1.pk3";
+    sha256 = "5610fd749024405b4425a7ce6397e58187b941d22092ef11d4844b427df53e5d";
+    };
+  pak2 = fetchAsset
+    { asset = "pak2.pk3";
+    sha256 = "a48ab749a1a12ab4d9137286b1f23d642c29da59845b2bafc8f64e052cf06f3e";
+    };
+  fakeGit = writeScriptBin "git" ''
+    #! ${stdenv.shell} -e
+    if [ "$1" = "describe" ]; then
+      if [ -r VERSION.txt ]; then
+        . <(sed 's/^\(VERSION_[A-Z]\+\) \(.*\)/\1=\2/' VERSION.txt)
+        VERSION="$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH"
+      fi
+      if [ "$VERSION" != "" ]; then
+        echo $VERSION
+      else
+        echo "Unable to determine version"
+        exit 1
+      fi
+    fi
+  '';
+  mainProgram = if stdenv.hostPlatform.system == "i686-linux" then "etl.i386" else "etl.x86_64";
+in stdenv.mkDerivation rec {
+  pname = pkgname;
+  inherit version;
+
+  src = fetchgit {
```

Comment:
to fetch from github, we recommand fetchFromGitHub because it downloads an archive instead of using git, which is more effecient. Also more robust to github changes.

Reply:
Switched to fetchFromGitHub